Back out c10 dispatcher diffs.

ezyang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#26277 Back out c10 dispatcher diffs.**

Back out "Fix binary size of OpsAlreadyMovedToC10.cpp"

Original commit changeset: c5420adf225e

Back out "Fix namedtensor ci"

Original commit changeset: aaef654953b4

Back out "Fix CI"

Original commit changeset: a87fb3048aeb

Back out "Remove unboxedAutogradKernel from c10"

Original commit changeset: d4468506a536

Back out "Call aten ops through c10 dispatcher"

Original commit changeset: 991f17b355e9

Differential Revision: [D17396585](https://our.internmc.facebook.com/intern/diff/D17396585/)

ezyang committed:
Back out c10 dispatcher diffs.

Back out "Fix binary size of OpsAlreadyMovedToC10.cpp"

Original commit changeset: c5420adf225e

Back out "Fix namedtensor ci"

Original commit changeset: aaef654953b4

Back out "Fix CI"

Original commit changeset: a87fb3048aeb

Back out "Fix Windows build"

Original commit changeset: 705998c0b160

Back out "Remove unboxedAutogradKernel from c10"

Original commit changeset: d4468506a536

Back out "Call aten ops through c10 dispatcher"

Original commit changeset: 991f17b355e9

Differential Revision: [D17396585](https://our.internmc.facebook.com/intern/diff/D17396585/)

smessmer reviewed (approved):
you're backing out too much, that's why CI on this PR is failing. My guess would be you have to remove "Fix Windows build" from the list to make it work. Btw there's probably no reason to back out "Fix binary size of OpsAlreadyMovedToC10.cpp" either.

ezyang committed:
Update on "Back out c10 dispatcher diffs."


Back out "Fix binary size of OpsAlreadyMovedToC10.cpp"

Original commit changeset: c5420adf225e

Back out "Fix namedtensor ci"

Original commit changeset: aaef654953b4

Back out "Fix CI"

Original commit changeset: a87fb3048aeb

Back out "Fix Windows build"

Original commit changeset: 705998c0b160

Back out "Remove unboxedAutogradKernel from c10"

Original commit changeset: d4468506a536

Back out "Call aten ops through c10 dispatcher"

Original commit changeset: 991f17b355e9

Differential Revision: [D17396585](https://our.internmc.facebook.com/intern/diff/D17396585/)

ezyang commented:
New failure is also legit:

```
15:12:57 terminate called after throwing an instance of 'c10::Error'
15:12:57   what():  For legacy aten ops, the schema() call must happen before any kernel() calls. Operator was aten::abs_(Tensor(a!) self) -> Tensor(a!) (schema at /var/lib/jenkins/workspace/aten/src/ATen/core/op_registration/op_registration.h:99)
15:12:57 frame #0: c10::Error::Error(c10::SourceLocation, std::__cxx11::basic_
```
